ludwig/models/predictor.py

```diff
@@ -270,18 +271,21 @@ def __init__(self, model: ECD, gpus=None, gpu_memory_limit=None, allow_parallel_
         self.batch_evaluation = return_first(self.batch_evaluation)
 
 
-def calculate_overall_stats(output_features, predictions, dataset, training_set_metadata):
+def calculate_overall_stats(output_features, predictions, dataset, training_set_metadata, df_engine):
     overall_stats = {}
     for of_name, output_feature in output_features.items():
         feature_metadata = training_set_metadata[output_feature.feature_name]
         feature_metadata.update(training_set_metadata[output_feature.feature_name])
 
         feature_df = predictions.loc[:, predictions.columns.str.startswith(of_name)]
         feature_df = feature_df.rename(columns=lambda c: c[len(of_name) + 1 :])
+        feature_df = feature_df.head(n=STATS_SAMPLE_SIZE, npartitions=-1, compute=True)
```

Comment:
I would only take the sample if the `feature_df` is a Dask DF. I would also log a warning stating "Sampling the first 10k rows from Dask DataFrame" or something to that effect.

Reply:
Done. I'm sampling in the case when `feature_df` is not a pandas Series, as I didn't want to add an extra dask import in `predictor.py`.